### PR TITLE
Dashboard friends

### LINKED ISSUE #11: Dashboard: Friends
As an authenticated user,
I see a section for friends,
In this section, there should be a text field to enter a friend's email and a button to "Add Friend"

**Scenarios**: 
1. If I have not added any friends there should be a message. "You currently have no friends".

2. If I have added friends, I should see a list of all my friends.

**Details:** Users should be able to add a friend by their email address, as long as, the friend is a user of our application and exists in our database. 

**Example:** 
Bugs Bunny and Lola Bunny are users of our application, but Daffy Duck is not. 
1. When Bugs Bunny enters `lola_bunny@gmail.com` to add friend it will be successful and Lola should show up as Bugs Bunny's friend.
2. When Bugs Bunny enters `daffy_duck@gmail.com` to add friend it should give an error message that the user does not exist. 

**Tips:** You'll want to research self-referential `has_many through`. [Here](http://blog.hasmanythrough.com/2007/10/30/self-referential-has-many-through) is a good starting point to understand the concept. You will probably need to do more googling but that's part of the fun ;)

- [x] Write a happy path test 
- [x] Write a sad path test

### PR DESCRIPTION
**Pull request by LHJE**
Closes #11 

This Pull Request:
- Adds to the dashboard a text field to type (hopefully) a new friend's email address.
- Adds a button that leads to Friendship Controller's create method
- Adds Friendship Controller
- Adds the Create method there
- Adds a route to get there.
- Adds tests for all this, including edge cases where if someone puts something that isn't an email address in the system, or if they put their own address, or if they put the address of a friend they already have.
- Cleans up a rogue extra line in _results_views file.
  
Notes:
- Rspec all passing
- Rubocop passing, except the new friendship create method, the old long line ... and another long one that is basically its duplicate plus some more.  See the files for that one. I've commented on it below too.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class FriendshipsController < ApplicationController
+  def create
+    new_friend = User.where(email: params["New Friend's Email"])
+    if new_friend == []
+      flash[:notice] = 'Friend not in our system.'
+    else
+      already_friend = User.where(id: Friendship.where(user_id: new_friend[0].id).pluck(:friend_id)).or(User.where(id: Friendship.where(friend_id: new_friend[0].id).pluck(:user_id))).where(id: current_user.id)
```
Comment (LHJE) [inline app/controllers/friendships_controller.rb:7]: This is basically a duplicate of the really long line we had before, but with `.where(id: current_user.id)` tacked on the end.  We could shorten this and the other one with .....

File: app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      if current_user.id == new_friend[0].id
+        flash[:notice] = "Well that's your email address! We do like that you're trying to be your own friend though :)"
+      elsif already_friend[0].nil?
+        Friendship.create(user_id: current_user.id, friend_id: new_friend[0].id)
+        # if this is what we have to do to get that other line longer, then we can implement this feature:
+        # Friendship.create(user_id: friend[0].id, friend_id: current_user.id)
```
Comment (LHJE) [inline app/controllers/friendships_controller.rb:13]: ... This extra line, as we have discussed.  I'm interested in seeing which of these two methods the instructors would prefer.

File: app/controllers/friendships_controller.rb (hunk 1 above)
Comment (PhilipDeFraties) [reply app/controllers/friendships_controller.rb:7]: Awesome job, we could probably have this live in the model

**Review by PhilipDeFraties — COMMENTED** [review]
Once the model method is made merge to dev????

**Review by PhilipDeFraties — APPROVED** [review]
